### PR TITLE
Merge old migrations

### PR DESCRIPTION
**Pull request by vpereira**
Even though the default way to deploy our application is via `rake db:schema:load` to have a broken `rake db:migrate` doesn't help. In this PR I did the following:

- Merged old migrations and deleted them
- Checked if all sql `INSERT` calls are covered by the `seeds.rb`
- Adapted old data migrations that were looking for old migration files
- Created a circleci job to check if any new introduced chages are breaking the migrations

I kept the last migration file there as placeholder, to make sure that we don't break production. After we create a new migration, we could safely delete this placeholder migration

Feel free to `rake db:migrate` your db as you wish :)

For a follow-up PR: 

- Clean the data migrations too

-----------------

If this PR requires any particular action or consideration before deployment,
please check the reasons or add your own to the list:

* [ ] Requires a database migration that can cause downtime. Postpone deployment until maintenance window[1].
* [ ] Contains a data migration that can cause temporary inconsistency, so should be run at a specific point of time.
* [ ] Changes some configuration files (e.g. options.yml), so the changes have to be applied manually in the reference server.
* [ ] A new Feature Toggle[2] should be enabled in the reference server.
* [ ] Proper documentation or announcement has to be published upfront since the introduced changes can confuse the users.

[1] https://github.com/openSUSE/open-build-service/wiki/Deployment-of-build.opensuse.org#when-there-are-migrations
[2] https://github.com/openSUSE/open-build-service/wiki/Feature-Toggles-%28Flipper%29#you-want-real-people-to-test-your-feature

### CODE REVIEW

File: src/api/spec/db/data/remove_duplicate_repositories_spec.rb
Comment (dmarcoux) [inline src/api/spec/db/data/remove_duplicate_repositories_spec.rb:3]: I think this is not the only thing which can be changed/removed in this file. See the comment on the lines 5 to 7 in this file. There's also some funky stuff happening in the _before_ and _after_ blocks below.

File: .circleci/config.yml
Comment (dmarcoux) [inline .circleci/config.yml:307]: I know it's not something we already do, but it could be nice to run this only when at least one file is changed in the directory `src/api/db`.

File: src/api/spec/db/data/remove_duplicate_repositories_spec.rb
Comment (vpereira) [reply src/api/spec/db/data/remove_duplicate_repositories_spec.rb:3]: yes, well it's for spec and the spec is passing, so i guess, we are good?  The question is if we really need those 4 years old data migrations?

Comment (dmarcoux) [reply src/api/spec/db/data/remove_duplicate_repositories_spec.rb:3]: It's good if the specs are already passing. If we keep the file around, I would still prefer to remove some of that weird code. It's going to confuse anybody looking into this.

It's true that we might not need those 4 years old data migrations anymore.

Comment (vpereira) [reply src/api/spec/db/data/remove_duplicate_repositories_spec.rb:3]: Yes, I would like to dig in the left over in another PR, otherwise the changelog will be even bigger. I think this task can be done in the "follow-up PR" as I described in the PR description.

**Comment by danidoni** [comment]
The `rake data:migrate` is still broken. Do you plan to work on this too?

**Comment by vpereira** [comment]
> The `rake data:migrate` is still broken. Do you plan to work on this too?

No, you just have to seed the DB, otherwise there is no data to migrate

If you look on the circleci config, I added a job to test the migrations, and we have the following there:

```rake db:drop && rake db:create && rake db:migrate && rake db:seed && rake data:migrate```

**Review by dmarcoux — APPROVED** [review]
Looking good, I gave it a try in my development environment and it works. Production will also be fine since we keep the current migration. Good job!

Please wait for another review before merging since this is quite a big change.

**Review by danidoni — APPROVED** [review]

**Pull request closed without merging**
